src/Data/Text/Encoding.hs

```diff
@@ -301,6 +296,131 @@ getCodePointStateOrError (Utf8DecodeState mCpSt (bs1@(B.length -> len1) : bss')
     Just (_, cpLen) -> Right cpLen
 getCodePointStateOrError _ = Right 0
 
+#ifdef SIMDUTF
+foreign import ccall unsafe "_hs_text_is_valid_utf8" c_is_valid_utf8
+    :: Ptr Word8 -> CSize -> IO CInt
+#endif
+
+{-
+parseUtf8ChunkFrom chunk from = (n, ms)
+
+n is the end index of the longest prefix of chunk that is valid UTF-8 starting
+from index 'from' where the length of the prefix = n - from
+
+ms
+* When ms = Nothing, there is an error: the bytes from index n and beyond are
+  not part of a valid UTF-8 code point.
+* When ms = Just s, all of the remaining bytes from index n and beyond are the
+  beginning of an incomplete UTF-8 code point, and s is the corresponding
+  intermediate decoding state, which can be used to parse the next chunk with
+  `parseUtf8Chunk`
+-}
+parseUtf8ChunkFrom :: ByteString -> Int -> (Int, Maybe Utf8CodePointState)
+parseUtf8ChunkFrom bs@(B.length -> len) pos =
```

Comment:
Can start from `0`, assuming the caller used `BS.drop pos` instead?

Reply:
Yes.